### PR TITLE
[7.0] Fix intellisense xml file selection from 'net' or 'dotnet-plat-ext' 

### PR DESCRIPTION
**Pull request by carlossanlop**
Addresses the 7.0 bug of https://github.com/dotnet/runtime/issues/82214

There are two places in 7.0 where the intellisense xml files are analyzed and/or copied.

- In packaging.targets, we decide if we want to include in the assembly package either the built xml or the one that comes from the internal intellisense nupkg. If we decide to use the latter, we were missing logic that would find the right xml file in either the 'net' or the 'dotnet-plat-ext' folder of the extracted nupkg. Instead, we were only looking inside 'net', and assemblies like System.IO.Ports were not getting anything included in its nupkg.

- In docs.targets, we were only copying to artifacts/bin/docs the files that were found in the 'net' folder of the internal intellisense nupkg. We need to also copy the files from 'dotnet-plat-ext', so that assemblies like System.IO.Ports get the right documentation shipped.

cc @krwq

### CODE REVIEW

File: eng/packaging.targets [hunk 1]
```diff
@@ -24,7 +24,10 @@
                                        '$(BuildAllConfigurations)' == 'true' and
                                        '$(DotNetBuildFromSource)' == 'true'">true</GeneratePackageOnBuild>
     <!-- Search for the documentation file in the intellisense package and otherwise pick up the generated one. -->
-    <LibIntellisenseDocumentationFilePath>$(XmlDocFileRoot)1033\$(AssemblyName).xml</LibIntellisenseDocumentationFilePath>
```
Comment (carlossanlop) [inline eng/packaging.targets:27]: `LibIntellisenseDocumentationFilePath` now gets conditionally defined in two places below.

`IntellisenseNetFile` is always processed, but only once (and used in two places).

`IntellisenseDotNetPlatExtFile` is conditionally processed once, only if we couldn't find the file under `IntellisenseNetFile` (and also used in two places).

File: eng/restore/docs.targets [hunk 2]
```diff
@@ -9,7 +9,7 @@
           AfterTargets="Restore">
 
     <ItemGroup>
-      <DocFile Include="$(XmlDocFileRoot)**\*.xml"/>
+      <DocFile Include="$(XmlDocFileRoot)\net\**\*.xml;$(XmlDocFileRoot)\dotnet-plat-ext\**\*.xml"/>
```
Comment (carlossanlop) [inline eng/restore/docs.targets:12]: I tried to include all the files under `$(XmlDocFileRoot)\*\**\*.xml` but it threw errors.

Instead, appending with a semicolon the two paths worked.

**Review by smasher164 — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]

**Comment by carlossanlop** [comment]
All CI failures have been logged above. They're unrelated.

@ViktorHofer if we want to properly fix the issue, we would have to enable `GeneratePackageOnBuild` and bump the `ServicingVersion` of all affected nuget packages.

I'd feel more comfortable doing this in a separate PR. Any suggestions/concerns about this? Same question would apply to the 6.0 version of this PR.

File: eng/restore/docs.targets (hunk 2 above)
Comment (carlossanlop) [reply eng/restore/docs.targets:12]: Talked offline with Viktor, he's ok with this.

**Comment by ViktorHofer** [comment]
We would need to re-ship quite a lot of packages so let's definitely not do this in this PR. IMO we should discuss this in one of the next Tactics meetings to get pre-approval before we even attempt to do the work.

**Comment by carlossanlop** [comment]
> We would need to re-ship quite a lot of packages so let's definitely not do this in this PR. IMO we should discuss this in one of the next Tactics meetings to get pre-approval before we even attempt to do the work.

Let's at least ship a new version of System.IO.Ports, since the issue started with that one. We can add the rest to the same PR depending on what Tactics says.

**Comment by carlossanlop** [comment]
Failures unrelated.

**Pull request merged**